### PR TITLE
[libnick] Update to 2024.1.13

### PR DESCRIPTION
**Pull request by nlogozzo**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by nlogozzo** [comment]
ok, that's the last of the updates...I promise :)

**Comment by nlogozzo** [comment]
@MonicaLiu0311 hope all is well! any reason for delay in review?

File: versions/l-/libnick.json [hunk 1]
```diff
@@ -1,5 +1,25 @@
...
+    {
+      "git-tree": "d71c91f939c5111259daf2c838da9c0094f0e80f",
+      "version": "2024.1.11",
+      "port-version": 0
+    },
+    {
+      "git-tree": "cc42a3022d565e282e3d48addb88acfb41b2bc04",
+      "version": "2024.1.10",
+      "port-version": 0
+    },
```
Comment (MonicaLiu0311) [inline versions/l-/libnick.json:22]:
```suggestion
```

Comment (MonicaLiu0311) [reply versions/l-/libnick.json:22]: - [ ] Only one version is added to each modified port's versions file.

Comment (nlogozzo) [reply versions/l-/libnick.json:22]: Ah got it. I'm actually about to release another version with some API changes I needed to make.

Sorry, first time writing a C++ library and using it in some apps 😅

I will open a new PR when ready

**Pull request closed without merging**